src/dynamo0p3.py

```diff
@@ -1209,33 +1266,174 @@ def initialise_dofmaps(self, parent):
+                parent.add(AssignGen(parent, pointer=True, lhs=dmap,
+                                     rhs=field.proxy_name_indexed +
+                                     "%" + field.ref_name() +
+                                     "%get_whole_dofmap()"))
+        if self._unique_cbanded_maps:
+            parent.add(CommentGen(parent, ""))
+            parent.add(CommentGen(parent,
+                                  " Look-up required column-banded dofmaps"))
+            parent.add(CommentGen(parent, ""))
+
+            for dmap, cma in self._unique_cbanded_maps.items():
+                parent.add(AssignGen(parent, pointer=True, lhs=dmap,
+                                     rhs=cma[0].proxy_name_indexed +
+                                     "%column_banded_dofmap_"+cma[1]))
+
+        if self._unique_indirection_maps:
+            parent.add(CommentGen(parent, ""))
+            parent.add(CommentGen(parent,
+                                  " Look-up required CMA indirection dofmaps"))
+            parent.add(CommentGen(parent, ""))
 
-        for dmap, field in self._unique_fs_maps.items():
-            parent.add(AssignGen(parent, pointer=True, lhs=dmap,
-                                 rhs=field.proxy_name_indexed +
-                                 "%" + field.ref_name() +
-                                 "%get_whole_dofmap()"))
+            for dmap, cma in self._unique_indirection_maps.items():
+                parent.add(AssignGen(parent, pointer=True, lhs=dmap,
+                                     rhs=cma[0].proxy_name_indexed +
+                                     "%indirection_dofmap_"+cma[1]))
 
     def declare_dofmaps(self, parent):
         ''' Declare all unique function space dofmaps as pointers to
         integer arrays of rank 2. The declarations are added as
         children of the supplied parent argument. This must be an
         appropriate f2pygen object. '''
         from f2pygen import DeclGen
+
+        # Function space dofmaps
         decl_map_names = \
             [dmap+"(:,:) => null()" for dmap in self._unique_fs_maps]
 
         if decl_map_names:
             parent.add(DeclGen(parent, datatype="integer", pointer=True,
                                entity_decls=decl_map_names))
 
+        # Column-banded dofmaps
+        decl_bmap_names = \
+            [dmap+"(:,:) => null()" for dmap in self._unique_cbanded_maps]
+        if decl_bmap_names:
+            parent.add(DeclGen(parent, datatype="integer", pointer=True,
+                               entity_decls=decl_bmap_names))
+
+        # CMA operator indirection dofmaps
+        decl_ind_map_names = \
+            [dmap+"(:) => null()" for dmap in self._unique_indirection_maps]
+        if decl_ind_map_names:
+            parent.add(DeclGen(parent, datatype="integer", pointer=True,
+                               entity_decls=decl_ind_map_names))
+
+
+class DynInvokeCMAOperators(object):
```

Comment:
I think it is better the way it is - the new dofmap entities fit well within the existing `DynInvokeDofmaps` class and they are conceptually different to the CMA operators themselves.